pkg/cmd/roachtest/roachtestutil/mixedversion/steps.go

```diff
@@ -230,8 +230,26 @@ func (s setClusterSettingStep) Description() string {
 func (s setClusterSettingStep) Run(
 	ctx context.Context, l *logger.Logger, rng *rand.Rand, h *Helper,
 ) error {
-	stmt := fmt.Sprintf("SET CLUSTER SETTING %s = $1", s.name)
-	return h.ExecWithGateway(rng, nodesRunningAtLeast(s.minVersion, h), stmt, s.value)
+	var stmt string
+	var args []interface{}
+	// We do a type switch on common types to avoid errors when using
+	// placeholders, as type detection is often not implemented for some
+	// private cluster settings.
+	switch val := s.value.(type) {
+	case string:
+		stmt = fmt.Sprintf("SET CLUSTER SETTING %s = '%s'", s.name, val)
```

Comment:
Nit: `%q`

Reply:
I've always seen strings in SQL being represented with single quotes? The PostgreSQL documentation says:

> A string constant in SQL is an arbitrary sequence of characters bounded by single quotes (')

And double quotes are used to represent database identifiers:

>  There is a second kind of identifier: the _delimited identifier_ or _quoted identifier_. It is formed by enclosing an arbitrary sequence of characters in double-quotes ("). [...] So "select" could be used to refer to a column or table named “select”, whereas an unquoted `select` would be taken as a key word and would therefore provoke a parse error when used where a table or column name is expected.

https://www.postgresql.org/docs/current/sql-syntax-lexical.html.

`%q` might just work here, but no harm being explicit.